pkg/plugins/aws/ami/condition.go

```diff
@@ -0,0 +1,74 @@
+package ami
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"github.com/updatecli/updatecli/pkg/core/scm"
+)
+
+// Condition test if a image matching specific filters exist.
+func (a *AMI) Condition(source string) (bool, error) {
+
+	if source != "" {
+		// Based on source information,
+		// we try to define a default image-id resource
+		// if not researched
+		isImageIDDefined := false
+		for i := 0; i < len(a.Filters); i++ {
+			if strings.Compare(a.Filters[i].Name, "image-id") == 0 {
+				isImageIDDefined = true
+				break
+			}
+
+		}
+
+		// Set image-id to source output if not yet defined
+		if !isImageIDDefined {
+			a.Filters = append(a.Filters, Filter{
+				Name:   "image-id",
+				Values: source,
+			})
+		}
+
+	} else {
+		if len(a.Filters) == 0 {
+			logrus.Infof("\u2717 No AMI could be found as no AMI filters defined\n")
+			return false, nil
+		}
+	}
```

Comment:
```suggestion
// It's an error if the upstream source is empty and the user does not provide any filter
// then it mean
if source == "" && len(a.Filters) == 0 {
	logrus.Infof("\u2717 No AMI could be found as no AMI filters defined\n")
	return false, nil
}

// Based on source information,
// we try to define a default image-id resource
// if not researched
isImageIDDefined := false
for i := 0; i < len(a.Filters); i++ {
	if strings.Compare(a.Filters[i].Name, "image-id") == 0 {
		isImageIDDefined = true
		break
	}

}

// Set image-id to source output if not yet defined
if !isImageIDDefined {
	a.Filters = append(a.Filters, Filter{
		Name:   "image-id",
		Values: source,
	})
}
```

As a general rule of thumb, try to return as early as possible. When you need to use a `else` then it might be a smell. The goal is to avoid overloading the code.

Reply:
(Check the gofmt 😅)